### PR TITLE
[release-4.3] Bug 1821318: feat(packagemanifest): add default os/arch labels if not present

### PR DESCRIPTION
**Pull request by shawn-hurley**
**Description of the change:**
Manual cherry-pick of: #1220

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/package-server/storage/subresources_test.go [hunk 1]
```diff
@@ -7,6 +7,8 @@ import (
 	"net/http/httptest"
 	"testing"
 
+	"k8s.io/apimachinery/pkg/labels"
```
Comment (ecordell) [inline pkg/package-server/storage/subresources_test.go:10]: this is a surprising diff - I would expect the build to fail

**Comment by ecordell** [comment]
the unit test failure looks legit and related to these changes

**Comment by sdodson** [comment]
@shawn-hurley Can you cherry-pick-approved this PR while I track down why I was removed from the operator-framework org?

**Pull request merged**